src/workerd/api/web-socket.h

```diff
@@ -642,12 +646,25 @@ class WebSocket: public EventTarget {
   struct GatedMessage {
     kj::Maybe<kj::Promise<void>> outputLock;  // must wait for this before actually sending
     kj::WebSocket::Message message;
+    size_t pendingAutoResponses = 0;
   };
   using OutgoingMessagesMap = kj::Table<GatedMessage, kj::InsertionOrderIndex>;
   // Queue of messages to be sent. This is wraped in a IoOwn so that the pump loop can safely
   // access the map without locking the isolate.
   IoOwn<OutgoingMessagesMap> outgoingMessages;
 
+  // Keep track of current hibernatable websockets auto-response status to avoid racing
+  // between regular websocket messages, and auto-responses.
+  struct AutoResponse {
+    kj::Promise<void> ongoingAutoResponse = kj::READY_NOW;
```

Comment:
Ahhhh, so I think what I was expecting was that `outgoingAutoResponse` would be a forked promise ala:
```
kj::Promise<void> ongoingAutoResponse = kj::Proimse<void>(kj::READY_NOW).fork();
```

I think what you have here works, it's just a touch unusual! I'm guessing it was hard to phrase `sendAutoResponse()` in that context.